packages/backend/src/dao/anonymous-id-dao.ts

```diff
@@ -0,0 +1,67 @@
+/*
+What is this file and does Polyratings track users?
+Due to administrative reasons we decided we needed a way to associate ratings and reports that were written by the same user. This was done for 
+multiple reasons:
+
+1. Mass delete a user that was trying to spam bad ratings for a professor
+2. Some people accidentally posted a rating in the wrong place and wanted it removed. Confirming this would be nice
+3. Disagreements over rating between individuals became a back and forth spamming of ratings and reports. It would be useful to disambiguate parties
+
+For these reasons we decided we needed to associate an identifier with each user of Polyratings that could correlate actions of a period of 
+multiple days. However, this identifier will ALWAYS be ANONYMOUS, and even we the developers no way to de-anonymize user. We decided to derive 
+this identifier from the users IP address but we NEVER store the IP directly anywhere. Instead we use the hash of the IP to associate it with
+a session id. In addition, since we are committed to open data we did not want to remove this identifier from the public Github dataset, we
+needed to make sure that session ids did not allow for a writing comparison de-anonymization. As such we evaluated two options for implementation:
```

Comment:
Glad we have this call out here. Nice to point out that we need to balance privacy considerations with data quality, but that privacy is always going to be our primary driver.